src/components/tab-panel/tab-panel.tsx

```diff
@@ -0,0 +1,93 @@
+import {
+    Component,
+    Element,
+    h,
+    Prop,
+    EventEmitter,
+    Event,
+    Host,
+} from '@stencil/core';
+import { Tab } from '../tab-bar/tab.types';
+
+/**
+ * @slot - Content to put inside the `tab-panel`. Each slotted element must have the `id` attribute equal to the id of the tab it belongs to.
+ */
+@Component({
+    tag: 'limel-tab-panel',
+    shadow: true,
+})
+export class TabPanel {
+    @Prop({ mutable: true })
+    public tabs: Tab[] = [];
+
+    @Element()
+    private host: HTMLElement;
+
+    @Event()
+    protected changeTab: EventEmitter<Tab>;
```

Comment:
Completely regardless of `protected` vs `private`, I think `eventEmitter` is more important than `host`, so I'd place `eventEmitter` above `host` 🙂

Reply:
Yeah I agree, I usually place the events after the props :+1: 

Interestingly though, I found this the other day and they have the `Element` almost above everything else :smile: 
https://stenciljs.com/docs/style-guide

I'm not saying that we should follow that 100%, but some of the points are quite interesting. Like placing the watchers right next to prop that is being watched for instance :slightly_smiling_face:

Reply:
And now I also found this in case we *do want* to enforce some rules :thinking: 
https://www.npmjs.com/package/tslint-stencil